### DISCUSSION TITLE
C++ Sample inputs for onnxruntime_perf_test.exe

### DISCUSSION THREAD

**Discussion by raoanagharaj** [microsoft/onnxruntime#5640] [Other Q&A]
I am trying to run onnxruntime_perf_test.exe with onnxruntime_perf_test.exe -v -e dml < onnxruntime\build\Debug\testdata\transform\matmul_add_fusion\3Input\model.onnx> path as input.
I also tried out other sample inputs in the repo, but mostly got the below error:

iteration:5,time_cost:0.0084346
**Run failed:PerformanceRunner::RunOneIteration caught exception: Unexpected input data type. Actual: (class onnxruntime::PrimitiveDataType<float>) , expected: (class onnxruntime::PrimitiveDataType<bool>)**

What are the best input data for onnxruntime_perf_test ?

**Comment by aawce** [comment]
Hi,

I too am trying to understand how to use onnxruntime_perf_test with my own known-good .onnx models.  We test the models in Keras with prediction data in .h5 files.  Are there any more detailed examples on how to use this onnxruntime_perf_test so I can get the data into the right format?

Thanks.

**Comment by snnn** [comment]
The model and test data format should be the same as the ones in ONNX model zoo. 
See:
https://github.com/onnx/models
